src/audio_core/hle/wmf_decoder_utils.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#ifndef MF_DECODER
+#define MF_DECODER
+
+#define WINVER _WIN32_WINNT_WIN7
+
+#include <assert.h>
+#include <comdef.h>
+#include <mfapi.h>
+#include <mferror.h>
+#include <mfidl.h>
+#include <mftransform.h>
+#include <stdio.h>
+
+#include <iostream>
+#include <string>
+
+#include "adts.h"
+
+// utility functions
+template <class T>
+void SafeRelease(T** ppT) {
```

Comment:
This could be avoided by using unique_ptr and would also "SafeRelease" in case of an exception